### PR TITLE
skip transport for another matrix based on error count

### PR DESCRIPTION
**Pull request by LSChyi**
Dynamically skip acquiring the other half matrix if there are transport errors.

## Description

The transport split hand causes the keyboard to become very laggy when the other half is not connected, down, or not yet ready. This is extremely serious when using the `SPLIT_USB_DETECT`, which is the default setting for Chibios, the other half hangs in the state until the `SPLIT_USB_TIMEOUT`. The default `SPLIT_USB_TIMEOUT` blocks the other hand and only starts working after 2 seconds, the main half is very laggy in the meanwhile and makes people think the keyboard is not working due to it tries to transport with the other half and times out all the time.

As a result, adding a mechanism to not acquiring the other half matrix based on the transport failure times could prevent the laggy behavior and still makes the main half working.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: common_features.mk [hunk 1]
```diff
@@ -428,8 +428,10 @@ ifeq ($(strip $(SPLIT_KEYBOARD)), yes)
         # Functions added via QUANTUM_LIB_SRC are only included in the final binary if they're called.
         # Unused functions are pruned away, which is why we can add multiple drivers here without bloat.
         ifeq ($(PLATFORM),AVR)
-            QUANTUM_LIB_SRC += i2c_master.c \
-                               i2c_slave.c
+            ifneq ($(NO_I2C),yes)
+                QUANTUM_LIB_SRC += i2c_master.c \
+                                   i2c_slave.c
+            endif
```
Comment (drashna) [inline common_features.mk:434]: Not really a fan of this change.  Especially as the way that these files are added, they should be optimized out, if they're not used.

**Comment by drashna** [comment]
Looks like you have some merge conflicts here, and looks like it needs to be rebased.

File: common_features.mk (hunk 1 above)
Comment (mtei) [reply common_features.mk:434]: I don't think this diff is part of this PR. This diff is a change made in #10204 that has already been merged.

Comment (LSChyi) [reply common_features.mk:434]: Sorry that this is because the base is force pushed - let me rebase it first. Actually, there shouldn't a change here.

**Comment by LSChyi** [comment]
Rebased - please have a look

**Comment by drashna** [comment]
Sorry, looks like there is a merge conflict here.

**Comment by tzarc** [comment]
Sorry about that, GitHub decided to delete the `develop` branch from upstream when we merged it, despite being told otherwise. Reopened.

**Pull request closed without merging**